### PR TITLE
[#3665]Clarified the use of glob pattern

### PR DESCRIPTION
**Pull request by svekars**
- Clarified the use of glob pattern.
- Basic editorial cleanup.

### CODE REVIEW

File: doc/reference/pipeline_spec.md [hunk 1]
```diff
@@ -173,99 +175,106 @@ In practice, you rarely need to specify all the fields.  Most fields either come
...
+`transform.image` is the name of the Docker image that your jobs use.
 
-`transform.cmd` is the command passed to the Docker run invocation.  Note that
-as with Docker, cmd is not run inside a shell which means that things like
-wildcard globbing (`*`), pipes (`|`) and file redirects (`>` and `>>`) will not
-work.  To get that behavior, you can set `cmd` to be a shell of your choice
-(e.g. `sh`) and pass a shell script to stdin.
+`transform.cmd` is the command passed to the Docker run invocation. Similarly
+to Docker, `cmd` is not run inside a shell which means that
+wildcard globbing (`*`), pipes (`|`), and file redirects (`>` and `>>`) does
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: should be "do not work"

File: doc/reference/pipeline_spec.md [hunk 2]
```diff
@@ -173,99 +175,106 @@ In practice, you rarely need to specify all the fields.  Most fields either come
...
+`pipeline.name` is the name of the pipeline that you are creating. Each
+pipeline needs to have a unique name. Pipeline names must meet the following
+prerequisites:
 
-- contain only alphanumeric characters, `_` and `-`
-- begin or end with only alphanumeric characters (not `_` or `-`)
-- be no more than 50 characters in length
+- Include only alphanumeric characters, `_` and `-`.
+- Begin or end with only alphanumeric characters (not `_` or `-`).
+- Do not exceed 50 characters in length.
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: bullet list should have the same structure:

Pipeline names must... include only...
Pipeline name must..begin or end with...
Pipeline name must... do not exceed... <-- doesn't match. 

remove the "do"?

File: doc/reference/pipeline_spec.md [hunk 3]
```diff
@@ -274,42 +283,43 @@ The default if left unset is "constant=1".
...
+workers in the system is too large, Kubernetes cannot schedule new
+workers because no machine has enough unclaimed memory. `cpu` works
 similarly, but for CPU time.
 
 By default, workers are scheduled with an effective resource request of 0 (to
 avoid scheduling problems that prevent users from being unable to run
-pipelines).  This means that if a node runs out of memory, any such worker
-might be killed.
+pipelines). This means that if a node runs out of memory, any such worker
+might be destroyed.
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: killed is actually the correct word here. When a node runs out of memory, the actual error message is "killed" and we call this OOM Killed (out of memory)

File: doc/reference/pipeline_spec.md [hunk 4]
```diff
@@ -429,24 +457,29 @@ Union inputs take the union of other inputs. For example:
...
+your code first determines which input directory is present. Starting
+with Pachyderm 1.5.3, we recommend that you give your inputs the
+same `Name`. That way your code only needs to handle data being present
+in that directory. This only works if your code does not need to be
 aware of which of the underlying inputs the data comes from.
 
-`input.union` is an array of inputs to union, note that these need not be
-`pfs` inputs, they can also be `union` and `cross` inputs. Although there's no
-reason to take a union of unions since union is associative.
+`input.union` is an array of inputs to a union. The inputs do not have to be
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: union is being used as a verb here, not a noun so remove "a" or perhaps we should not use that verb, although I dont have an alternative in mind.

File: doc/reference/pipeline_spec.md [hunk 5]
```diff
@@ -457,22 +490,25 @@ tuples of the datums in the inputs. For example:
...
-unix utility `cron`. When you create a pipeline with one or more Cron Inputs
-pachd will create a repo for each of them. When a cron input triggers,
-that is when the present time satisfies its spec, pachd will commit
-a single file, named by the current [RFC
-3339 timestamp](https://www.ietf.org/rfc/rfc3339.txt) to the repo which contains the time which
-satisfied the spec.
+Cron inputs allow you to trigger pipelines based on time. A Cron input is
+based on the Unix utility called `cron`. When you create a pipeline with
+one or more Cron Inputs, `pachd` creates a repo for each of them. When
+a cron input triggers, that is when the present time satisfies its spec,
```
Comment (JoeyZwicker) [inline doc/reference/pipeline_spec.md]: > , that is when the present time satisfies its spec,

This phrase here is a bit kludgy IMO. Perhaps change:
, that is when the present time satisfies its spec, --> (e.g. when the present time satisfies its spec)

File: doc/reference/pipeline_spec.md (hunk 4 above)
Comment (svekars) [reply doc/reference/pipeline_spec.md]: combine?

Comment (JoeyZwicker) [reply doc/reference/pipeline_spec.md]: yeah combine is great.

**Pull request merged**